go/vt/vttablet/tabletmanager/rpc_external_reparent.go

```diff
@@ -160,9 +161,14 @@ func (agent *ActionAgent) finalizeTabletExternallyReparented(ctx context.Context
 		}
 	}()
 
+	tablet := agent.Tablet()
+	tabletMap, err := agent.TopoServer.GetTabletMapForShard(ctx, tablet.Keyspace, tablet.Shard)
+	// make the channel buffer big enough that it doesn't block senders
+	tablets := make(chan topodatapb.Tablet, len(tabletMap))
```

Comment:
Actually, we might not technically need a channel at all. Since we bail out if any errors occur, we should get the same result if we append tablets to a slice before launching each per-tablet goroutine (before we know whether the tablet record update will succeed, but after we decided to try to update it), which removes the need for synchronization. However, I'm fine keeping this as a channel if you prefer, since the proof that it isn't needed is a bit obscure.

Reply:
I was being cautious here, because UpdateTabletFields reads the latest tablet info from topo, and updates and returns it. This ensures that we are using the latest tabletInfo, but maybe it doesn't matter?

Reply:
Ah I see what you mean. I don't think it matters in this case since I believe `tmc.RefreshState` only looks at the tablet host/port, and we aren't changing those fields. There's a slight possibility the tablet might update its own host/port in between when we first read it and when we call RefreshState, but that window will always be non-zero and it doesn't get much smaller by taking the result from UpdateTabletFields, since we call those concurrently right after we fetched the latest values from GetTabletMapForShard.

That said, the above is yet another non-obvious nuance in the proof that we don't need synchronization, so it's probably better to just keep the synchronization to be defensive against future code changes.